### PR TITLE
Memory mapped data loading

### PR DESCRIPTION
**Pull request by mlgill**
This PR accomplishes two things:

* Renames the `io.py` file to `io_local.py` and fixes the associated imports to avoid a collision with the Python3 library of the same name
* Changes the data loading to use Keras `HDF5Matrix` which is memory mapped. This is an otherwise drop-in replacement for loading all the data into memory. I have tested both changes.

Happy to answer any questions if you're interested in using this PR. Thanks.

### CODE REVIEW

**Comment by tkschmidt** [comment]
Hey :)
I love it and will try to do some testing on the weekend. Do you have any numbers on how much this speeds up loading the data?

**Comment by mlgill** [comment]
Hi!

I wasn't able to time it the way it was originally written with all the data because the desktop station I am using ran out of RAM -- I couldn't load all of it in memory.

There is a possibility that it might be a little slower the second way, depending on how Keras manages memory for the dataset. But I doubt it will be significant.

So the biggest advantage is more efficient memory usage. Let me know what you find though.

**Comment by mlgill** [comment]
Hi -- Following up with two updates:

I ran a comparison of the total training time for 10 epochs on 500k samples for the old and new data loading methods. This includes data loading time for both so that it's an apples-to-apple comparison. Here is the wall time for each: 
* Old version (all data loading before training): 455 s total, 43.3 s / epoch for training
* New version (data loaded dynamically): 442 s total,  43.0 s / epoch for training

So , with ~1/10th of the data, there is negligible difference per epoch between loading the data into memory before running vs dynamically loading of the drive. And the second method requires slightly less training time overall (no data loading step at the beginning) and less memory. 

Many of Keras data loading methods are smart about pre-caching results and I suspect the time for contiguous reads from the disk is pretty short, so I suspect that's why the second method is still competitive even when compared to loading all data at the start.

These results can vary based on data size and drive I/O speed, but I thought this would be a good quick benchmark.

**Comment by gessulat** [comment]
Thank you so much @mlgill, this looks great!
